java-to-crd/src/main/java/io/javaoperatorsdk/crd/JavaToCrdAnnotationProcessor.java

```diff
@@ -0,0 +1,115 @@
+package io.javaoperatorsdk.crd;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import com.google.auto.service.AutoService;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionBuilder;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionVersionBuilder;
+import io.fabric8.kubernetes.client.utils.Pluralize;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Kind;
+import io.fabric8.kubernetes.model.annotation.Plural;
+import io.fabric8.kubernetes.model.annotation.Singular;
+import io.fabric8.kubernetes.model.annotation.Version;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.StandardLocation;
+
+@SupportedAnnotationTypes("io.javaoperatorsdk.crd.CRD")
+@SupportedSourceVersion(SourceVersion.RELEASE_11)
+@AutoService(Processor.class)
+public class JavaToCrdAnnotationProcessor extends AbstractProcessor {
+
+  private static final YAMLMapper mapper = new YAMLMapper();
+  private static ProcessingEnvironment env;
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    env = processingEnv;
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    try {
+      for (TypeElement annotation : annotations) {
+        Set<? extends Element> annotatedElements = roundEnv.getElementsAnnotatedWith(annotation);
+        annotatedElements.stream().map(e -> (TypeElement) e).forEach(this::generateCRD);
```

Comment:
The  `generateCRD` method only makes sense for `TypeElement` annotated with `CRD`.
Currently this is only enforced by the value of `@SupportedAnnotationTypes`. As, this is something that will definitely change moving forward it might be a good idea to actually guard it with code.

Reply:
I'll cross that bridge when I get there but yes, this is something to keep in mind. That said, right now, I envision requiring the `CRD` annotation because I'm not sure what the alternative would be (unless using extension of the `CustomResource` class as a criterium?).